webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/nodes/NodeInfoService.scala

```diff
@@ -267,18 +269,129 @@ trait NodeInfoServiceCached extends NodeInfoService with NamedZioLogger with Cac
    * attributes is the list of attributes needed in returned entries
    */
   def getNodeInfoEntries(con: RoLDAPConnection, attributes: Seq[String], status: InventoryStatus): LDAPIOResult[Seq[LDAPEntry]]
-
+  def getNewNodeInfoEntries(
+                             con: RoLDAPConnection
+                           , lastKnowModification: DateTime
+                           , searchAttributes: Seq[String]
+                           ): LDAPIOResult[Seq[LDAPEntry]]
 
   override def getNumberOfManagedNodes: Int = nodeCache.map(_.managedNodes).getOrElse(0)
 
   /*
    * Our cache
    */
-  private[this] var nodeCache = Option.empty[LocalNodeInfoCache]
+  protected[this] var nodeCache = Option.empty[LocalNodeInfoCache]
 
   // we need modifyTimestamp to search for update and entryCSN to remove already processed entries
   private[this] val searchAttributes = nodeInfoAttributes :+ A_MOD_TIMESTAMP :+ "entryCSN"
 
+  import scala.collection.mutable.{Map => MutMap}
+
+  // goes over each 3 maps to ensure the consistency of the maps
+  // for each updated node, we need to look for matching nodeInventories and machineInventories
+  // same for the other two
+  // and construct relevant LDAPNodeInfo
+  def constructNodes(
+         nodes              : MutMap[String, LDAPEntry]
+       , nodeInventories    : MutMap[String, LDAPEntry]
+       , machineInventories : MutMap[String, LDAPEntry]): Seq[LDAPNodeInfo] = {
+
```

Comment:
please add `currentNodeCache` as a parameter: it wil make the logic of the method clearer, and it will make it testable in isolation